### ISSUE TITLE
CollectionPropertiesShouldBeReadOnly (CA2227) reports on DataMembers

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#948]
#### Analyzer package
Microsoft.ApiDesignGuidelines.Analyzers

#### Analyzer
CA2227: CollectionPropertiesShouldBeReadOnly

#### Repro steps
CA2227 is reported on the `Gates` property in the next class.

```csharp
    [DataContract(Namespace = "", Name = "NetworkConfiguration")]
    public sealed class NetworkConfigurationXml
    {
        [DataMember]
        public Collection<GateSettingsXml> Gates { get; set; }
    }
```

#### Expected behavior
In this case, the serialization framework requires a public setter. It would be nice to special-case methods that are decorated with `DataMemberAttribute` or similar serialization annotations.

#### Actual behavior
CA2227 is reported on methods that are decorated with `DataMemberAttribute`.

**Comment by mavasani** [comment]
@bkoelman Once again thanks for the contribution. For future, it would be helpful if you can also add a comment on issue that you are working on, so it can avoid possible duplicate work (especially if it already has an assignee). Thanks!

**Comment by bkoelman** [comment]
Good to know, I will do that next time.
Op 10 mei 2016 1:53 a.m. schreef "Manish Vasani" <notifications@github.com>:
